static/app/components/button.tsx

```diff
@@ -526,6 +534,7 @@ const StyledButton = styled(
     shouldForwardProp: prop =>
       prop === 'forwardRef' ||
       prop === 'external' ||
+      prop === 'replace' ||
       (typeof prop === 'string' && isPropValid(prop)),
   }
```

Comment:
this was honestly hard for me to find - I was just seeing the `replace` prop not being forwarded correctly. Imo this “StyledButton” does way too many things that aren’t styling related at all.

Reply:
100% agree. I would like for us to untangle this, it's way too complex for a button :D